neo/Network/P2P/Peer.cs

```diff
@@ -129,25 +137,31 @@ private void OnStart(int port, int ws_port)
                     localAddresses.Add(UPnP.GetExternalIP());
                     if (port > 0)
                         UPnP.ForwardPort(port, ProtocolType.Tcp, "NEO");
-                    if (ws_port > 0)
-                        UPnP.ForwardPort(ws_port, ProtocolType.Tcp, "NEO WebSocket");
+                    if (wsPort > 0)
+                        UPnP.ForwardPort(wsPort, ProtocolType.Tcp, "NEO WebSocket");
                 }
                 catch { }
             }
             if (port > 0)
             {
                 tcp_manager.Tell(new Tcp.Bind(Self, new IPEndPoint(IPAddress.Any, port), options: new[] { new Inet.SO.ReuseAddress(true) }));
             }
-            if (ws_port > 0)
+            if (wsPort > 0)
             {
-                ws_host = new WebHostBuilder().UseKestrel().UseUrls($"http://*:{ws_port}").Configure(app => app.UseWebSockets().Run(ProcessWebSocketAsync)).Build();
+                ws_host = new WebHostBuilder().UseKestrel().UseUrls($"http://*:{wsPort}").Configure(app => app.UseWebSockets().Run(ProcessWebSocketAsync)).Build();
                 ws_host.Start();
             }
         }
 
         private void OnTcpConnected(IPEndPoint remote, IPEndPoint local)
         {
             ImmutableInterlocked.Update(ref ConnectingPeers, p => p.Remove(remote));
+            if (ConnectedPeers.Count >= MaxConnections && !TrustedIpAddresses.Contains(remote.Address))
```

Comment:
Good for those nice guys :dagger: :dancer:

Reply:
I’ll make it allow using -1 for unlimited

Reply:
I understand that this PR just allow to use `Connect` with `Trusted` argument. But be careful in the use, never is removed a Trusted Ip, and with IPV6 is easy to fill your memory with only one address (if is used wrong).

Is possible to add a parameter `IsTrusted` here (`OnTcpConnected`) and remove `TrustedIpAddress` list ?

Reply:
No. it may become disconnected and get reconnected in the normal connection loop. It is not meant ot be turning on the flag on random addresses or anything; it should only be turned on for a well defined set of trusted addresses. This should never be taking much memory. I could make it have a max if we wanted, but I think it is an unnecessary protection.